numpy/ma/core.py

```diff
@@ -2418,12 +2418,13 @@ def _recursive_printoption(result, mask, printopt):
 
     """
     names = result.dtype.names
-    for name in names:
-        (curdata, curmask) = (result[name], mask[name])
-        if curdata.dtype.names:
+    if names:
```

Comment:
technically not necessary since this function is always called with names (or else the old code would fail). Fine to leave it.

Reply:
OK, I get that this was intentional based on reorganization below.